parquet-variant-compute/src/variant_array.rs

```diff
@@ -968,6 +964,42 @@ fn typed_value_to_variant<'a>(
         DataType::Float64 => {
             primitive_conversion_single_value!(Float64Type, typed_value, index)
         }
+        DataType::Decimal32(_, s) => {
+            generic_conversion_single_value!(
+                Decimal32Type,
+                as_primitive,
+                |v| VariantDecimal4::try_new(v, *s as u8).unwrap(),
+                typed_value,
+                index
+            )
+        }
+        DataType::Decimal64(_, s) => {
+            generic_conversion_single_value!(
+                Decimal64Type,
+                as_primitive,
+                |v| VariantDecimal8::try_new(v, *s as u8).unwrap(),
+                typed_value,
+                index
+            )
+        }
+        DataType::Decimal128(_, s) => {
+            generic_conversion_single_value!(
+                Decimal128Type,
+                as_primitive,
+                |v| VariantDecimal16::try_new(v, *s as u8).unwrap(),
```

Comment:
Are we _sure_ these `try_new` calls will never fail? 

I mean, we _are_ unshredding shredded data, but it's also... data... which is _not_ trustworthy.

For safety, suggest to do something like:

```suggestion
                |v| VariantDecimal16::try_new(v, *s as u8).map_or(Variant::Null, Variant::from),
```

Reply:
Thanks for this suggestion, fixed.